cmplibrary/src/test/java/com/sourcepoint/cmplibrary/util/extensions/CcpaCSExtKtTest.kt

```diff
@@ -0,0 +1,185 @@
+package com.sourcepoint.cmplibrary.util.extensions
+
+import com.sourcepoint.cmplibrary.assertEquals
+import com.sourcepoint.cmplibrary.data.network.model.optimized.CcpaCS
+import com.sourcepoint.cmplibrary.model.exposed.CcpaStatus
+import org.junit.Test
+
+internal class CcpaCSExtKtTest {
+
+    /**
+     * Test case that covers the case when:
+     * applies == null
+     * then the consent string should be == "1---"
+     */
+    @Test
+    fun `generateConsentString - WHEN called with applies as null THEN should return default CCPA consent string`() {
+        // GIVEN
+        val ccpaCS = CcpaCS(
+            actions = null,
+            applies = null,
+            ccpaApplies = null,
+            consentedAll = null,
+            cookies = null,
+            dateCreated = null,
+            gpcEnabled = null,
+            newUser = null,
+            rejectedAll = null,
+            rejectedCategories = null,
+            rejectedVendors = null,
+            signedLspa = false,
+            status = null,
+            uspstring = null,
+            uuid = null,
+        )
+        val expected = "1---"
+
+        // WHEN
+        val actual = ccpaCS.generateConsentString()
+
+        // THEN
+        actual.assertEquals(expected)
```

Comment:
For brevity, we could substitute:
```kotlin
val expected = "1---"

// WHEN
val actual = ccpaCS.generateConsentString()

// THEN
actual.assertEquals(expected)
```
with
```kotlin
ccpaCS.generateConsentString().assertEquals("1---")
```

In my opinion it keeps things shorter and still easy to understand what's going on (specially if we use that helper method I mentioned above)

Reply:
from my pov, if we take an abstact term of what is the purpose of the test, then it is to match the `actual` result to the `expected` one

in this case, I also wanted to make sure that `generated` result right from the helper class `SpUtils` generates the same value, as we get from the `actual`

that is why, from my pov, the original way is more readable, because you can see what are `actual` and `expected` results